modules/vsearch/cluster/main.nf

```diff
@@ -0,0 +1,89 @@
+process VSEARCH_CLUSTER {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::vsearch=2.21.1 bioconda::samtools=1.15.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-53dae514294fca7b44842b784ed85a5303ac2d80:6e91f82b987ac1c6b6a24e292a934ae41b17311d-0':
+        'quay.io/biocontainers/mulled-v2-53dae514294fca7b44842b784ed85a5303ac2d80:6e91f82b987ac1c6b6a24e292a934ae41b17311d-0' }"
+
+    input:
+    tuple val(meta), path(fasta)
+    val clusteroption
+    val idcutoff
+    val outoption
+    val user_columns
```

Comment:
Is there a reason not to pass them vie the modules.config as usual for values? :)

Reply:
All these arguments are mandatory, this is why I add them as inputs. Except for `user_columns` which is only mandatory if `outoption == userout`. Do you think it will be better to add `user_columns` in modules.config?

Reply:
yes, this is how we treat all modules so far (with rare exceptions of course) , only paths are passed over but all arguments (even mandatory once) are passed over the config. (only argument that is set explicetly that is not a file is generally the `task.cpus` one.

Reply:
ok, I have moved the parameters to the config file but still maintained the checks. I used the same format for ext.args as the module `ascat`, hope it's fine :)

Reply:
Why do you need the "ascat" way? :) We had this weird setup thre only because it is actually an Rscript where you can't pass the args over as a string as you would normal do:

```
tool.py --param1 xyz --param2 qwe
```

etc. but it is setup more like this:

```
someRfunction(param1)
anotherRfunction(param2, someRfunction.output)
```

Reply:
🤔 I would like to test that the provided arguments are correct or otherwise use defaults, so I was looking for a way to get them from `ext.args`. Is there another way to do that?

Reply:
Checking on this again. I think that the arguments `clusteroption` and `outoption` need to be passed as inputs, I need them to provide the input fasta and output file name. This is the help command from the doccumentation:
`vsearch (--cluster_fast | --cluster_size | --cluster_smallmem | --cluster_unoise) fastafile (--alnout | --biomout | --blast6out | --centroids | --clusters | --mothur_shared_out | --msaout | --otutabout | --profile | --samout | --uc | --userout) outputfile --id real [options]`
but please let me know if there is a different way to do that that I am not aware of :)
What I removed are the other parameters, as you suggested.

Could you have a second look please? Thank you very much!

Reply:
solved 😄